### PR TITLE
[Browser] Update ICU

### PR DESCRIPTION
**Pull request by EgorBo**
new ICU package now supports multiple *.dat files, see https://github.com/dotnet/icu/pull/20

### CODE REVIEW

File: src/libraries/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs [hunk 1]
```diff
@@ -176,6 +176,11 @@ public void TestFirstYearOfJapaneseEra()
             DateTime dt = new DateTime(1989, 01, 08); // Start of Heisei Era
 
             string formattedDateWithGannen = "\u5E73\u6210 \u5143\u5E74 01\u6708 08\u65E5";
+            if (PlatformDetection.IsBrowser)
+            {
+                formattedDateWithGannen = "Heisei \u5143\u5E74 01\u6708 08\u65E5";
```
Comment (EgorBo) [inline src/libraries/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs:181]: in the newer ICU data we had to give up on native names for calendars but probably we need undo it for japanese one.

**Comment by steveisok** [comment]
@lewing Should we backport this?

File: src/mono/wasm/wasm.targets [hunk 2]
```diff
@@ -63,7 +63,7 @@
     <Copy SourceFiles="$(NativeBinDir)dotnet.js;
                        $(NativeBinDir)dotnet.wasm;
                        $(NativeBinDir)dotnet.timezones.blat;
-                       $(NativeBinDir)icudt.dat"
+                       $(NativeBinDir)icudt_optimal.dat"
```
Comment (safern) [inline src/mono/wasm/wasm.targets:66]: Should we add a property that can be set for the filter name and then copy the according .dat file?

Comment (EgorBo) [reply src/mono/wasm/wasm.targets]: @safern ah, indeed, actually it should copy all of the *.dat files to the runtime -pack
fixed.

File: eng/liveBuilds.targets [hunk 3]
```diff
@@ -192,7 +192,7 @@
         $(LibrariesNativeArtifactsPath)dotnet.js;
         $(LibrariesNativeArtifactsPath)dotnet.wasm;
         $(LibrariesNativeArtifactsPath)dotnet.timezones.blat;
-        $(LibrariesNativeArtifactsPath)icudt.dat;"
+        $(LibrariesNativeArtifactsPath)icudt_optimal.dat;"
```
Comment (akoeplinger) [inline eng/liveBuilds.targets:195]: this also needs to copy all *.dat files

**Review by akoeplinger — CHANGES_REQUESTED** [review]

File: eng/liveBuilds.targets (hunk 3 above)
Comment (EgorBo) [reply eng/liveBuilds.targets]: fixed via wildcard

**Review by akoeplinger — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Pull request merged**